app/src/main/java/org/airella/airella/StartActivity.kt

```diff
@@ -3,32 +3,61 @@ package org.airella.airella
 import android.content.Intent
 import android.os.Bundle
 import androidx.appcompat.app.AppCompatActivity
+import androidx.constraintlayout.motion.widget.MotionLayout
+import androidx.drawerlayout.widget.DrawerLayout
 import org.airella.airella.data.service.AuthService
 import org.airella.airella.data.service.PreferencesService
 import org.airella.airella.ui.login.LoginActivity
 import org.airella.airella.utils.Log
 
 class StartActivity : AppCompatActivity() {
-
+    var launched = false
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
+        setContentView(R.layout.activity_start)
+
         PreferencesService.init(applicationContext)
 
-        setContentView(R.layout.activity_start)
+        val motionLayout: MotionLayout = findViewById(R.id.motion_layout)
+        val context = applicationContext;
+        motionLayout.setTransitionListener(object : MotionLayout.TransitionListener {
+            override fun onTransitionCompleted(p0: MotionLayout?, p1: Int) {
+
+            }
+
+            override fun onTransitionChange(
+                p0: MotionLayout?,
+                p1: Int,
+                p2: Int,
+                animationPercent: Float
+            ) {
+                if (animationPercent > 0.8 && !launched) {
+                    launched = true
+                    if (AuthService.isUserLogged()) {
+                        Log.i("User logged")
+                        val intent = Intent(context, MainActivity::class.java)
```

Comment:
you should create new Intent using activity context, not application context, use "this" instead ("this@StartActivity" in this example)